### DISCUSSION TITLE
Designer can't load assembly in VS 2022 for .NET Framework

### DISCUSSION THREAD

**Discussion by RayCarrot** [MahApps/MahApps.Metro#4226] [Q&A] [answered]
After updating to Visual Studio 2022 it appears that the XAML designer no longer works for this library. In my app.xaml where I have the resources specified it says `Could not load file or assembly 'MahApps.Metro, Culture=neutral' or one of its dependencies. The system cannot find the file specified.` for all MahApps.Metro resource dictionaries. Inside of every other .xaml file the designer is showing the default Windows styles rather than the MahApps.Metro ones.
I can still compile and when running the build all the styles appear correctly, so it's only the designer that doesn't pick it up. I've noticed this across all of my projects which target .NET Framework while my projects which target .NET Core are all fine. Since this seems to consistently happen across multiple of my projects I don't think it's an issue there. Is this happening for anyone else? If not, does anyone have any ideas what the issue could be?
Thanks!

**Comment by fredericoregateiro** [comment]
Hello @RayCarrot 
I'm using VS2022 now with no problem, but when i started i had a error in one style that caused only problems in VS2022 not in VS2019.
Can you post a sample with your problem?

**Comment by RayCarrot** [comment, marked as answer]
After a bunch of testing I figured out it apparently only happens in projects where I use `Costura.Fody`. As soon as I remove its reference (or simply remove it from the `FodyWeavers.xml` file) then it works fine.
It's also worth noting that if I include the resource dictionaries directly in the xaml file I'm editing then the designer loads them fine. It only gives me an error when they're only defined in my app.xaml.
I'm at a bit of a loss about what to do though since I never experienced this issue before. I still believe it's related to VS22 since it started around the time I upgraded. I tried downgrading Costura to see if it was an update there which broke it, but that made no difference.
I would also imagine this isn't directly related to MahApps.Metro then and rather an issue with resources defined in separate assemblies, so sorry if this discussion is out of place here.

Edit: I found a temporary solution. If I have Costura only be applied conditionally when in Release mode then the designer will work in Debug mode. This works well enough for now since I only need Costura when publishing.
